### PR TITLE
Added translationally invariant stabilizer codes

### PR DESCRIPTION
**Pull request by Tonexus**
For IQC QEC course, Anthony Lau. It's not quite clear what the correct way to specify relationship with the Kitaev surface code is, as the surface code seems like it should be a "child" of both CSS codes and translationally invariant codes, but neither CSS nor translationally invariant imply the other while a code can only have 1 parent.

### CODE REVIEW

**Comment by valbert4** [comment]
A code can have multiple parents :).

**Comment by Tonexus** [comment]
Great!

File: codes/quantum/qubits/translationally_invariant_stabilizer.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: translationally_invariant_stabilizer
+physical: qubits
+logical: qubits
+
+name: 'Translationally invariant stabilizer code'
```
Comment (valbert4) [inline codes/quantum/qubits/translationally_invariant_stabilizer.yml:10]: Translationally-invariant stabilizer code

File: codes/quantum/qubits/translationally_invariant_stabilizer.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+introduced: '\cite{arXiv:1607.01387}'
```
Comment (valbert4) [inline codes/quantum/qubits/translationally_invariant_stabilizer.yml:11]: arXiv:1101.1962,arXiv:1204.1063
Add notes: field and mention that arXiv:1607.01387 and Sec. IV of arXiv:1812.01625 are good introductions into the correspondence between Laurent polynomials and TI stabs.

File: codes/quantum/qubits/translationally_invariant_stabilizer.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+description: |
+  A family of stabilizer codes in which the qubits are organized on a lattice modeled by the additive group \(\mathbb{Z}^D\) for
+  spatial dimension \(D\), each lattice point contains a finite number \(q\) qubits, and the stabilizer group is translationally
+  invariant with respect to the lattice.
```
Comment (valbert4) [inline codes/quantum/qubits/translationally_invariant_stabilizer.yml:16]: Mention that this makes the codes geometrically local.

File: codes/quantum/qubits/translationally_invariant_stabilizer.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: translationally_invariant_stabilizer
+physical: qubits
```
Comment (valbert4) [inline codes/quantum/qubits/translationally_invariant_stabilizer.yml:7]: This is a solid entry; thanks for the effort. Let's make this a "property" meaning it transcends kingdoms, like qldpc or topological_abelian. I've made a yml file for this in quantum/properties/qldpc in order to make the proper cousin/parent relations. Please update that file next round.

File: codes/quantum/qudits_galois/translationally_invariant_prime_qudit_stabilizer.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#######################################################
```
Comment (valbert4) [inline codes/quantum/qudits_galois/translationally_invariant_prime_qudit_stabilizer.yml:1]: Please merge this with TI stab above.

File: codes/quantum/qudits_galois/translationally_invariant_prime_qudit_stabilizer.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  qubits, and the stabilizer group is translationally invariant with respect to the lattice.
+  
+  Similar to the qubit case, a single-qudit Pauli operator can be specified by the lattice coordinate of the site and the
+  \(\mathbb{F}_p^{2q}\) symplectic vector representation of the Pauli operator within the site. The stabilizer group of the
+  translation invariant code is then generated by site-local Pauli operators and their translations. As each lattice coordinate can
+  be represented by a Laurent monomial of \(D\) formal variables, the multiplicative group of finitely supported Pauli operators modulo
+  phase factors on the lattice of dimension \(D\) with \(q\) qubits per site is isomorphic to the additive group of Laurent polynomial
+  column vectors of length \(2q\) in \(D\) formal variables.
+
+relations:
```
Comment (valbert4) [inline codes/quantum/qudits_galois/translationally_invariant_prime_qudit_stabilizer.yml:26]: Keep parents/cousins in the original file I just made.

File: codes/quantum/qubits/translationally_invariant_stabilizer.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  multiple copies of the 3D toric code.
+  
+  2. Foliated type-I: Excitations are mobile in less than 3 dimensions, but codes can be grown by "foliation"—stacking copies of the 2D
+  toric code.
+  
+  3. Fractal type-I: Excitations are mobile in less than 3 dimensions, and codes are not foliated.
+  
+  4. Type-II: Excitations are not mobile in any dimension.
+
+relations:
```
Comment (valbert4) [inline codes/quantum/qubits/translationally_invariant_stabilizer.yml:45]: Keep parents/cousins in the original file I just made. Re D=0 case "reducing" to stabilizers, I'm not to keen on having that be a parent distinction; do you feel strongly about this? I propose we put this idea that one can think of stabilizer codes as 0D TI stabs in the description of this class.

**Pull request closed without merging**